### ISSUE TITLE
Bwcheck_httpstreaming plugin does not relay custom NetStream events

### ISSUE THREAD

**Issue by hmatanock** [flowplayer/flash#280]
We are injecting a custom NetStream event into live streams we are serving off WOWZA servers - events that tell the player to roll mid-roll ads.  If we use the static version of our stream, these events are relayed through the httpstreaming plugin no problem.  However, when we use mulit-bitrate adaptive streams, the bwcheck_httpstreaming plugin does not relay the events.

I believe that I've figured out the fix for this and I can submit a pull request.  The real problem I'm having is getting everything to compile and still have the bandwith checking work correctly.  I've pulled a clean copy of everything you have here, compiled all three plugins (f4m, bwcheck and httpstreaming) and it kind of works.  I can playback the stream, and the bwcheck plugin will switch up to the higher bitrate for a minute, but then it switches back down and stays there.  Whereas the version of the plugins you have for download on your site work correctly.

Would you either be willing to compile the plugins for me using the modified code, or can you advise me on what I'm doing wrong to compile these?  The only thing I can thing of is that there is a file in the bwcheck plugin that references particular AS classes with local paths on your machine:

https://github.com/flowplayer/flash/blob/master/plugins/bwcheck/lib/externs.xml

perhaps I need to have those as well?  I'm also using the 4.8 Flex SDK, do you use a different version? 

Thank you in advance for any assistance you can offer

**Comment by danrossi** [comment]
thanks for that. so its all working ok for you or not ?  you need to run ant -f build-httpstreaming.xml it's in a seperate build.

**Comment by hmatanock** [comment]
Right, i know about the different ant xml, I can get it to compile, and it loads fine, without any errors, and the stream event passes through, that's all good.  The problem is that the adaptive stream switching stops working.  It will switch up to the highest bitrate for a second, then switch back down to the lowest and stay there.  If i view the exact same stream with the version of the plugin downloaded from the flowplayer website, the bitrate switching works great, switching up and staying at the higher bitrate.

I'm thinking that I'm missing some classes or have the wrong versions of classes, although I've started totally over with a clean clone of your repositories and it still doesn't work for me.

Thanks!
Heather

**Comment by danrossi** [comment]
Use the normal one for RTMP The httpstreaming one is for HDS. No idea unless I see a config sorry. I use the same features for my HLS feature and its fine.

**Comment by danrossi** [comment]
And it needs the right osmf branch, I think its supplied in the github also. Try and remove the "osmf.swc" from your flex directory also. that might cause problems.

**Comment by danrossi** [comment]
Is this working for you at all ?

**Comment by hmatanock** [comment]
Yes.  Is it not for you?  You'd have to inject a custom NetStream event at the server. (or maybe listen for one that isn't usually handled by flowplayer, not sure which that would be)  You'd also need to register the stream event name as a streamCallback parameter with the main player.  See http://flash.flowplayer.org/documentation/configuration/player.html

I basically lifted this code change out of the main player's library and put it here because this plugin replaces the standard NetStreamClient with a custom one.  Our custom server events are handled correctly if I do not use this plugin, so it works.
https://github.com/flowplayer/flash/blob/master/core/src/actionscript/org/flowplayer/controller/NetStreamClient.as

I'm fine with maintaining my own custom mod of this plugin if you want to reject this pull request, I just need to get it to compile correctly.  I didn't get a chance to test removing the osmf.swc from my flex directory yesterday.  Will be testing that today.
